planner/core/collect_column_stats_usage.go

```diff
@@ -271,9 +273,17 @@ func (c *columnStatsUsageCollector) collectFromPlan(lp LogicalPlan) {
 }
 
 // CollectColumnStatsUsage collects column stats usage from logical plan.
-// The first return value is predicate columns and the second return value is histogram-needed columns.
-func CollectColumnStatsUsage(lp LogicalPlan) ([]model.TableColumnID, []model.TableColumnID) {
-	collector := newColumnStatsUsageCollector(collectPredicateColumns | collectHistNeededColumns)
+// predicate indicates whether to collect predicate columns and histNeeded indicates whether to collect histogram-needed columns.
+// The first return value is predicate columns(nil if predicate is false) and the second return value is histogram-needed columns(nil if histNeeded is false).
+func CollectColumnStatsUsage(lp LogicalPlan, predicate, histNeeded bool) ([]model.TableColumnID, []model.TableColumnID) {
```

Comment:
Why not pass the one parameter 'mode uint64' directly rather than two parameters `predicate, histNeeded bool`?